.github/workflows/bundleTest-thor.yml

```diff
@@ -0,0 +1,238 @@
+                          popd
+                      fi
+                  done
+              fi
+              
+              if [[ ! "${BUNDLES_TO_TEST[*]}" =~ "$bundleName"  ]]
+              then
+                  continue
+              fi
+              
+              pushd $bundleRunPath
+          
+              /opt/HPCCSystems/testing/regress/ecl-test run -t thor --config /opt/HPCCSystems/testing/regress/ecl-test.json --timeout 3600 -fthorConnectTimeout=3600 --pq 1 --ef ClassicTestModified.ecl,SVCTest.ecl,ClassificationTestModified.ecl
+              
+              retCode=$( echo $? )
+          
+              if [ ${retCode} -eq 0 ] 
+              then
+                  ProcessLog "$bundleName" "thor"
+              fi
+            
+              popd
+  
+          done< <(find . -iname 'ecl' -type d | sort )
+      - name: Generate ZAP files
+        if: ${{ ! inputs.Generate-Zap == '' }} 
+        run: |  
+          IFS=' ' read -a ML_GENERATE_ZAP_FOR <<< ${{ inputs.Generate-Zap }}
+          if [  ${#ML_GENERATE_ZAP_FOR[@]} -ne 0 ]
+          then
+              for test in ${ML_GENERATE_ZAP_FOR[*]}
+              do 
+                  wuid=$(ecl getwuid -n $test --limit 1)
+                  if [ -n $wuid ]
+                  then
+                      ecl zapgen $wuid  --path /home/runner/HPCCSystems-regression/zap --inc-thor-slave-logs
+                  fi
+              done
+          fi
+
+      - name: Check for Core files
+        run: |
+          NUM_OF_ML_CORES=( $(sudo find /var/lib/HPCCSystems/ -iname 'core*' -mtime -1 -type f -exec printf "%s\n" '{}' \; ) )
+          
+          GDB_CMD='gdb --batch --quiet -ex "set interactive-mode off" -ex "echo \n Backtrace for all threads\n==========================" -ex "thread apply all bt" -ex "echo \n Registers:\n==========================\n" -ex "info reg" -ex "echo \n Disas:\n==========================\n" -ex "disas" -ex "quit" '
+
+          if [ ${#NUM_OF_ML_CORES[@]} -ne 0 ]
+          then      
+              for  core in ${NUM_OF_ML_CORES[@]}
+              do
+                  base=$( dirname $core )
+                  lastSubdir=${base##*/}
+                  comp=${lastSubdir##my}
+          
+                  #sudo gdb --batch --quiet -ex "set interactive-mode off" -ex "thread apply all bt" -ex "quit" "/opt/HPCCSystems/bin/${comp}" $core | sudo tee "$core.trace" 2>&1
+                  sudo ${GDB_CMD} "/opt/HPCCSystems/bin/${comp}" $core | sudo tee "$core.trace" 2>&1
+                  cp "$core.trace" /home/runner/HPCCSystems-regression/log/ 
+            done
+          fi
+
+      - name: Get test stat
+        if: ${{ inputs.Get-Stat }}
+        run: |
+          ./QueryStat2.py -p /home/runner/HPCCSystems-regression/log/  -d '' -a --timestamp --compileTimeDetails 1 --graphTimings --allGraphItems --addHeader
```

Comment:
If this part is running, the generated .csv and .cfg files should add to artifact and upload.